setup.py

```diff
@@ -39,13 +39,8 @@
 # Set up install for Cython or Command line interface
 if os.environ.get('CLIB') != 'CMD_LINE':
 
-    # Check that compiler is set
-    if os.environ.get('CC') not in ['gcc','icc','clang','msvc'] and re.findall('gcc',str(os.environ.get('CC')))==[]:
-        warnings.warn('CC environment variable not set to valid value. Using default CC=gcc.')
-        os.environ["CC"] = 'gcc'
-
-    # OpenMP gcc compile: tested for MacOS and Linux
-    if os.environ.get('CC') =='gcc' or re.findall('gcc',str(os.environ.get('CC')))!=[]:
+    # Check for compiler env variable. If not set, assume it's a gcc build (linux & macOS only).
+    if os.environ.get('CC') not in ['icc','clang','msvc']:
```

Comment:
While it's quite a bit more painful to implement, instead of assuming `gcc` if's not in a short list of other options, perhaps it would be safer try to run `$CC --version`, and check the result for the string "gcc"?

Reply:
Not sure about that. It would rely on the shell env variable CC being set. If it's not set we want it to default to the gcc compiler options anyway.  

Those checks on 'CC' are just to modify the compiler options for those in the short list.  So I don't see a problem with letting them default to ones that work for gcc.

Reply:
Good point; agreed.